src/main/java/nextstep/qna/domain/Question.java

```diff
@@ -1,92 +1,58 @@
 package nextstep.qna.domain;
 
+import nextstep.qna.CannotDeleteException;
 import nextstep.users.domain.NsUser;
 
 import java.time.LocalDateTime;
 import java.util.ArrayList;
-import java.util.List;
 
-public class Question {
-    private Long id;
+public class Question extends BaseEntity{
 
-    private String title;
+    private QuestionMetaData questionMetaData;
 
-    private String contents;
-
-    private NsUser writer;
-
-    private List<Answer> answers = new ArrayList<>();
-
-    private boolean deleted = false;
-
-    private LocalDateTime createdDate = LocalDateTime.now();
-
-    private LocalDateTime updatedDate;
-
-    public Question() {
-    }
+    private final Answers answers = new Answers(new ArrayList<>());
 
     public Question(NsUser writer, String title, String contents) {
         this(0L, writer, title, contents);
     }
 
     public Question(Long id, NsUser writer, String title, String contents) {
         this.id = id;
-        this.writer = writer;
-        this.title = title;
-        this.contents = contents;
-    }
-
-    public Long getId() {
-        return id;
+        questionMetaData = new QuestionMetaData(writer, title, contents);
     }
 
-    public String getTitle() {
-        return title;
-    }
-
-    public Question setTitle(String title) {
-        this.title = title;
-        return this;
-    }
-
-    public String getContents() {
-        return contents;
-    }
-
-    public Question setContents(String contents) {
-        this.contents = contents;
-        return this;
-    }
-
-    public NsUser getWriter() {
-        return writer;
+    public Question(Long id, QuestionMetaData questionMetaData) {
+        this.id = id;
+        this.questionMetaData = questionMetaData;
     }
 
     public void addAnswer(Answer answer) {
         answer.toQuestion(this);
         answers.add(answer);
     }
 
-    public boolean isOwner(NsUser loginUser) {
-        return writer.equals(loginUser);
+    @Override
+    public String toString() {
+        return "Question [id=" + id + " " + questionMetaData + " ]";
     }
 
-    public Question setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
-    }
+    private void validateDeletable(NsUser loginUser) throws CannotDeleteException {
+        if (!questionMetaData.isOwner(loginUser)) {
+            throw new CannotDeleteException("질문을 삭제할 권한이 없습니다.");
+        }
 
-    public boolean isDeleted() {
-        return deleted;
+        answers.validateDeletable(loginUser);
     }
 
-    public List<Answer> getAnswers() {
-        return answers;
+    public DeleteHistories delete(NsUser loginUser) throws CannotDeleteException {
+        validateDeletable(loginUser);
+        deleted = true;
```

Comment:
```suggestion
        deleted();
```

좀 더 메시지를 전달해본다는 관점으로 접근해보면 어떨까요?
`delete`함수는 검증도하고 값도 변경하고, DeleteHistories 도 만들고 복잡해보이는데,

각각을 함수의 메시지로 표현하면 분리된 함수 안에서는 단일책임원칙, 순수함수를 지킬수 있지 않을까요?